### PR TITLE
#50 enable export of timezone from metadata in to-mos

### LINKED ISSUE #50: export time zone - customizable or similar to import
**Is your feature request related to a problem? Please describe.**
Some expect the export to be the same time zone as the import file.
This is only relevant for .mos or .epw exports.

Currently all exports are in the time zone UTC. 
This is in line with the core data, which will always be converted to UTC. It is also step toward standardization and the lean option. As e.g. UTC+1, might arise the question whether it includes summer and winter time, as it might be just taken over from the import data.

Though some tools that use these weather files might not work as expected as they do not read the time zone. Additionally, it might be an unexpected result if the data is shifted. Also, if importing exactly one year in e.g. UTC+2, the export in UTC will have two hours at the end of the data set which are filled by ffill. 

**Describe the solution you'd like**
Option 1: Keep it like is
Option 2: Always use the time zone of the import (if available)
Option 3: Give the choice to the user
Option 4: A mix of the prior.

**Solution proposal**
To solve this, I suggest saving (or defining) the time zone in the meta data object, which is already done in some cases, and take it from there during any core_data_2_export_file transformation. The fix should actually be simple. 
Though, if the user should be able to set the time zone by oneself, the webapp would need an update as well.

Also important: update documentation in readme and within code

### PR DESCRIPTION
**Pull request by FWuellhorst**
This closes #50.

Todos:
- [x] Check correctnes
- [x] Add test
- [x] add epw output and possibly others

### CODE REVIEW

**Review by MartinRaetz — CHANGES_REQUESTED** [review]
Thank you very much @FWuellhorst.

Can you confirm that the tests will cover sufficiently the new feature? Since its the new default the regression tests should be running on the new default.

**Comment by FWuellhorst** [comment]
@MartinRaetz:  I checked locally in Dymola for a single test and the feature works well for Modelica.

@TobiasSpratte : Can you check the with E+ if the exported .epw with the timezone also works as expected, especially in combination with SPAWN and mos. I don't know how to access the weather results from E+ in Spawn.

Future development options: 
- [ ] Add a simple test model to at least check created mos files locally for different tilts and orientations, see screenshot
image

- [ ] Make all tests check both utc and export with timezone
- [ ] add an example with a different timezone (e.g. -6) to trigger edge cases for large shifting

What do you think @MartinRaetz ? I would do that in other issues to keep this one focused on the timezone aspect.

**Comment by TobiasSpratte** [comment]
I have checked the exported .mos and .epw files in EnergyPlus, Spawn and Modelica and everything works as expected. Thank you @FWuellhorst !

File: aixweather/core_data_format_2_output_file/to_epw_energyplus.py [hunk 1]
```diff
@@ -460,6 +466,9 @@ def _format_data(df, start, stop, fillna):
         df, start, stop
     )
 
+    ### Shift to desired timezone
+    df = df.shift(periods=timezone, freq="h", axis=0)
```
Comment (MartinRaetz) [inline aixweather/core_data_format_2_output_file/to_epw_energyplus.py:470]: Gehe davon aus, dass das klappt. Aber achte drauf, ob nach dem shift der Datensatz noch zum "richtigen" Zeitpunkt startet. Glaube für modelica und try wichtig, sonst nicht.

File: aixweather/imports/DWD.py [hunk 2]
```diff
@@ -208,6 +208,7 @@ def import_meta_DWD_historical(station: str) -> utils_import.MetaData:
     meta.station_exists_since = station_data["BEGINN"]
     meta.station_exists_until = station_data["ENDE"]
     meta.input_source = "DWD Historical"
+    meta.timezone = 1  # Always in Germany
```
Comment (MartinRaetz) [inline aixweather/imports/DWD.py]: Bin mir hier nicht sicher. 1 ist ja nur winterzeit. Die Daten die man beim DWD lädt, waren (glaube ich) UTC. 
Erklär hier mal die Wirkung der Zeile, ansonsten müsste die Timezone auch in dem "dict" stehen. Ist aber alles bisschen was her.

File: aixweather/imports/ERC.py [hunk 3]
```diff
@@ -126,6 +126,7 @@ def import_meta_from_ERC() -> MetaData:
     meta.altitude = 230  # ca. 200m + building height 30m
     meta.latitude = 50.7893
     meta.longitude = 6.0516
+    meta.timezone = 1
```
Comment (MartinRaetz) [inline aixweather/imports/ERC.py]: hier gleiche frage. Oder ist die timezone nur für den export relevant, weil in meta?

Comment (MartinRaetz) [reply aixweather/imports/ERC.py]: Ah, glaube das ist alles nur die standard spec für den Export, richtig? Wenn das so ist passt!

**Review by MartinRaetz — APPROVED** [review]
Thank you for this major PR! I think its a very helpful modification to meet the users expectations!

File: aixweather/imports/ERC.py (hunk 3 above)
Comment (FWuellhorst) [reply aixweather/imports/ERC.py]: Ja genau, nur bei epw wird das auch für den import genutzt. Habe ich im Kommentare hinzugefügt.

File: aixweather/core_data_format_2_output_file/to_epw_energyplus.py (hunk 1 above)
Comment (FWuellhorst) [reply aixweather/core_data_format_2_output_file/to_epw_energyplus.py:470]: Ich füge in diesem branch auch schon mos-simulationen hinzu, dann sieht man direkt den Effekt.

**Comment by FWuellhorst** [comment]
@MartinRaetz @TobiasSpratte 
I added regression tests which simulate the generated time-shifted mos in Modelica on different tilts and directions. You can create these and more plots with the new test_mos_simulation, executed as a plain python script locally. 

## Tilted roof first day of year diffuse radiation
plots_summer=False_HDifTilRoo

## Tilted roof in summer direct radiation
plots_summer=True_HDirTilRoo


Overall, when exporting time zones shifts are valid. Shifting larger than 6-8 hours compared to the imported data, the values become higher in the plot, which is caused by the values of the next day in the original data. 
Still, I stored the _imported_timezone (which is set in the import classes with "set_imported_timezone" and added a warning if someone changes the default timezone of the import.
